File: scode/DRAGONS/prob11/main.py
s = input()
if(len(s)!=0):
    n=int(s)
    l=[]
    for i in range(n):
      l1=input().split(" ")
      #print(l1)
      l.append(l1)

    main=[x for x in l if x[1]=="Main"]
    print(main)
    A = [x for x in l if x[1]=="A"]
    print(A)
    B = [x for x in l if x[1]=="B"]
    print(😎

    M=[]
    U=[]
    N=[]
    for i in A:
        t=True
        for j in main:
          if (i[2:]==j[2:]):
             N.append(i[2:])
          else:
              for k in j[2:]:
                  for l in i[2:]:
                      if (k==l):
                          t=False
                          break
              if(t):
                  U.append(i[2:])
              else:
                  M.append(i[2:])

    A=B
    for i in A:
        for j in main:
          if (i[2:]==j[2:]):
              N.append(i[2:])
          else:
              for k in j[2:]:
                  for l in i[2:]:
                      if (k != l):
                          t = False
                          break
              if (t):
                  N.append(i[2:])
              else:
                  U.append(i[2:])

    for i in M:
        for j in i:
         print(j,end=' ')
        print(": M")
    M=U
    #print(M)
    for i in M:
      for j in i:
         print(j,end=' ')
      print(": U")
    M=N
    for i in M:
        for j in i:
            print(j,end=' ')
        print("")
else :
    print("NO")

File: scode/DRAGONS/prob11/main.cpp
#include<bits/stdc++.h>

using namespace std;
typedef long long ll;
typedef unsigned long long ull;

int main()
{
    ios_base::sync_with_stdio(0);cin.tie(0);
    vector<int> input;
    string s;
    bool test=true;
    getline(cin,s);
    for(int i=0;i<s.size();i++)
    {
        if(isdigit(s[i]))
        {
            if(!isdigit(s[i+1])){
                int a=s[i]-'0';
                input.push_back(a);
            }
            else
            {
                int a=s[i]-'0';
                int b=s[i+1]-'0';
                input.push_back(a*10+b);
                break;
            }
        }
        if(isalpha(s[i]))
            test=false;
    }
    if(test){
    int sum=0;
    for(int i=0;i<input.size();i++)
    {
        sum+=input[i];
    }
    int k=0;
    int num=sum;
    //cout<<num<<endl;
    while(num)
    {
        num=num/2;
        k++;
    }
    //cout<<k<<endl;
    //cout<<sum<<endl;
    int l=k;
    while(k)
    {
        //cout<<k<<endl;
        //cout<<j<<endl;
        int j=0;
        if(sum>>(k-1)&1)
            j = 1<<k-1;
        //cout<<j<<endl;
        string ch="";
        if(j==1)
            ch="take_position";
        else if(j==2)
            ch="run";
        else if(j==4)
            ch="need_backup";
        else if(j==8)
            ch="declare_your_position";
        if(k==l&&j)
            cout<<ch;
        else if(j)
            cout<<' '<<ch;
        k--;
    }
    }
    else
        cout<<"NOT_CLEAR"<<endl;
}
